tests/test_taskgroups.py

```diff
@@ -542,6 +542,28 @@ async def cancel_when_ready():
     assert not sleep_completed
 
 
+async def test_shielding_mutate():
+    async def cancel_on_event(cancel_scope: CancelScope, event: anyio.Event):
+        await event.wait()
+        cancel_scope.cancel()
+
+    completed = False
+    async with create_task_group() as tg:
+        with CancelScope() as scope:
+            event = anyio.Event()
+            tg.start_soon(cancel_on_event, tg.cancel_scope, event)
+            await wait_all_tasks_blocked()
+            scope.shield = True
+            event.set()
+            await sleep(.1)
+            completed = True
+            scope.shield = False
+            await sleep(0)
+            assert False
+
+    assert completed
```

Comment:
A more concise test for this. Passes on trio.
```suggestion
async def test_shielding_mutate():
    async def task(*, task_status):
        with CancelScope(shield=True) as scope:
            task_status.started()
            await sleep(.1)
            scope.shield = False
            await sleep(1)
            pytest.fail('Execution should not reach this point')

    async with create_task_group() as tg:
        await tg.start(task)
        tg.cancel_scope.cancel()
```

Reply:
The coverage is not quite equivalent, as the original code was also confirming that a shield transition to True from the body of the cancel scope is honored (and to make extra sure, not immediately on entrance of the scope).

I used Event to minimize reliance on wall time.

Reply:
I've adopted your solution but added use of fail_after() to avoid a long sleep in the normal case

Reply:
These changes completely defeat the purpose of the test. That `sleep(1)` was supposed to be cancelled due to `shield=False`. Now the task has no more checkpoints where it could be cancelled. Was that the long sleep you referred to?

Reply:
> Now the task has no more checkpoints where it could be cancelled.

I believe the test is correct.  See comments below.

```python
  async def task(task_status):
        with CancelScope() as scope:
            await sleep(.1)
            scope.shield = True
            task_status.started()
            await sleep(.1)  # Parent's start() call returns, at which point it issues cancel().
                             # This sleep() completes because we're shielded.
            scope.shield = False
            done.set()

    with fail_after(1):
        async with create_task_group() as tg:
            await tg.start(task)
            tg.cancel_scope.cancel()
            await done.wait()
```

Reply:
I see-- there was a bug that done.wait() would not happen because the tg was cancelled.

Fixed that and confirmed that the test fails if `scope.shield = True` is commented out.